test/rpc/validation.js

```diff
@@ -0,0 +1,59 @@
+const test = require('tape')
+const request = require('supertest')
+
+const { startRPC, closeRPC } = require('./helpers')
+const { middleware } = require('../../lib/rpc/validation')
+
+test('should comlete the request with non-params if the method does not require any params', t => {
+  const server = startRPC({
+    'mock': middleware((params, cb) => {
+      cb()
+    }, 0, [])
+  })
+  const req = {
+    jsonrpc: '2.0',
+    method: 'mock',
+    id: 1
+  }
+
+  request(server)
+    .post('/')
+    .set('Content-Type', 'application/json')
+    .send(req)
+    .expect(res => {
+      if (res.body.error) {
+        throw new Error('should not return an error object')
+      }
+    })
+    .end(err => {
+      closeRPC(server)
+      t.end(err)
+    })
+})
+
+test('should throw if the method requires params but non-params', t => {
+  const server = startRPC({
+    'mock': middleware((params, cb) => {
+      cb()
+    }, 1, [])
+  })
+  const req = {
+    jsonrpc: '2.0',
+    method: 'mock',
+    id: 1
+  }
+
+  request(server)
+    .post('/')
+    .set('Content-Type', 'application/json')
+    .send(req)
+    .expect(res => {
+      if (!res.body.error) {
+        throw new Error('should return an error object')
```

Comment:
There is also not an explicit "what should have happened?" case here, we should think about that.

Currently a request like `curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x1b4", true],"id":1}' http://127.0.0.1:8545` is actually crashing the whole client, this should likely be fixed? 😄 

Ouput:

```shell
ERROR [06-08|12:38:02] uncaughtException: Cannot read property 'length' of undefined
TypeError: Cannot read property 'length' of undefined
    at Eth.getBlockByNumber (/ethereumjs-client/lib/rpc/validation.js:12:18)
    at Method.execute (/ethereumjs-client/node_modules/jayson/lib/method.js:123:20)
    at /ethereumjs-client/node_modules/jayson/lib/server/index.js:303:12
    at maybeParse (/ethereumjs-client/node_modules/jayson/lib/server/index.js:428:5)
    at Server.call (/ethereumjs-client/node_modules/jayson/lib/server/index.js:231:3)
    at /ethereumjs-client/node_modules/jayson/lib/utils.js:151:14
    at /ethereumjs-client/node_modules/lodash/lodash.js:10050:25
    at /ethereumjs-client/node_modules/jayson/lib/utils.js:117:7
    at Object.Utils.JSON.parse (/ethereumjs-client/node_modules/jayson/lib/utils.js:254:3)
    at IncomingMessage.<anonymous> (/ethereumjs-client/node_modules/jayson/lib/utils.js:113:16)
```

What is the default/specified behavior here?

Reply:
Update: ah sorry, got a bit confused here. Just realizing that I  just tested on my local `master` and that this case WAS actually fixed by your, PR, lol (that's what I was referring too with "learning/work out together" in the other post 😄 ).

Anyhow, so the type of test seems ok here. This might also take an additional `expect(200)` here, but not too pressing.

Reply:
So the output that this actually gives is:

```shell
HTTP/1.1 200 OK
Connection: keep-alive
Date: Mon, 08 Jun 2020 12:13:37 GMT
content-length: 100
content-type: application/json; charset=utf-8

{
    "error": {
        "code": -32602,
        "message": "missing value for required argument 0"
    },
    "id": "1",
    "jsonrpc": "2.0"
}
```

Just for the sake of completeness.

Reply:
Thank you for trying this by yourself!